balancer/rls/picker.go

```diff
@@ -252,6 +252,25 @@ func (p *rlsPicker) handleRouteLookupResponse(cacheKey cacheKey, targets []strin
 	// entry would be used until expiration, and a new picker would be sent upon
 	// backoff expiry.
 	now := time.Now()
+
+	// "An RLS request is considered to have failed if it returns a non-OK
+	// status or the RLS response's targets list does not contain any non-empty
+	// entries." - RLS LB Policy design.
+	allTargetsEmpty := true
+	for _, target := range targets {
+		if target != "" {
+			allTargetsEmpty = false
+			break
+		}
+	}
+	if allTargetsEmpty {
+		if err == nil {
+			err = fmt.Errorf("RLS response's target list does not contain any non-empty entries for key %+v", cacheKey)
+		}
```

Comment:
Let's move this `if err == nil` check to surround all of this new logic?

Reply:
Done, alongside implementing offline discussion wrt just checking length of targets slice.

Reply:
Updated description.